### PR TITLE
#164046236 Update Django to version 2.1

### PR DESCRIPTION
**Pull request by cob04**
#### What does this PR do?
Upgrades from Django 1.10 to Django 2.1.
#### Description of Task to be completed?
Update Django to version 2.1 plus all libraries that have Django as a dependency. 
#### How should this be manually tested?
* Clone this repo `git clone -b origin/ch-migrate-to-django2.1-164046236 https://github.com/andela/ah-legion-backend.git`
* Install dependencies with the command `pip install -r requirements.txt` from the root folder.
* Run `python manage.py runserver`
#### Any background context you want to provide?
Moved the codebase to Python3, because Django 2.1 only supports Python3
#### What are the relevant pivotal tracker stories?
[#164046236](https://www.pivotaltracker.com/story/show/164046236)

### CODE REVIEW

File: authors/apps/authentication/renderers.py [hunk 1]
```diff
@@ -18,7 +18,6 @@ def render(self, data, media_type=None, renderer_context=None):
             # rendering errors.
             return super(UserJSONRenderer, self).render(data)
```
Comment (MandelaK) [inline authors/apps/authentication/renderers.py]: While I believe this is to support backwards compatibility with python2.7, Django no longer supports python2.7, so I don't think it is necessary here. What do you think?

**Comment by Etomovich** [comment]
When we upgrade django to 2.1 in the requirements.txt file are we to uninstall the dependencies of django 1.1 or all can be there to enable apps written with django 1.1 to run despite of the upgrade. Color will be highly appreciated. Thanks.

**Review by Etomovich — COMMENTED** [review]
When we upgrade django to 2.1 in the requirements.txt file are we to uninstall the dependencies of django 1.1 or all can be there to enable apps written with django 1.1 to run despite of the upgrade. Color will be highly appreciated. Thanks.

**Comment by cob04** [comment]
@Etomovich  I have updated all the dependencies that are affected by upgrading Django.  
Hope that addresses your concern. Thank you.

File: authors/apps/authentication/renderers.py (hunk 1 above)
Comment (cob04) [reply authors/apps/authentication/renderers.py]: You are right, let me update that to support only Python3

**Comment by abulojoshua1** [comment]
> When we upgrade django to 2.1 in the requirements.txt file are we to uninstall the dependencies of django 1.1 or all can be there to enable apps written with django 1.1 to run despite of the upgrade. Color will be highly appreciated. Thanks.

@Etomovich, upgrading the Django version will mean that we need to do away with all Django 1.10 dependencies and only install Django 2.1 dependencies. One app cannot run on two Django versions. Also, `https://pypi.org/` keeps new versions and older versions of dependencies because some apps may still be running on older dependency versions and this cuts across all dependency repositories irrespective on programing language.

**Review by MandelaK — APPROVED** [review]
Everything looks good to me

**Comment by Etomovich** [comment]
> > When we upgrade django to 2.1 in the requirements.txt file are we to uninstall the dependencies of django 1.1 or all can be there to enable apps written with django 1.1 to run despite of the upgrade. Color will be highly appreciated. Thanks.
> 
> @Etomovich, upgrading the Django version will mean that we need to do away with all Django 1.10 dependencies and only install Django 2.1 dependencies. One app cannot run on two Django versions. Also, `https://pypi.org/` keeps new versions and older versions of dependencies because some apps may still be running on older dependency versions and this cuts across all dependency repositories irrespective on programing language.

Thanks. It is now clear.

**Comment by RachelleMaina** [comment]
Everything looks great

**Review by RachelleMaina — APPROVED** [review]
Everything looks fine.

**Review by Etomovich — APPROVED** [review]
Everything looks good to go.

**Review by ElMonstro — APPROVED** [review]
It good to go

**Comment by BrianSerem** [comment]
Looks good, a good start

**Review by BrianSerem — APPROVED** [review]
Looks good, its a good start

**Review by abulojoshua1 — APPROVED** [review]
LGTM

**Pull request merged**